joblib/numpy_pickle.py

```diff
@@ -334,25 +254,14 @@ def load_build(self):
         dispatch[pickle.BUILD] = load_build
 
 
-class ZipNumpyUnpickler(NumpyUnpickler):
-    """A subclass of our Unpickler to unpickle on the fly from
-    compressed storage."""
-
-    def __init__(self, filename, file_handle):
-        NumpyUnpickler.__init__(self, filename,
-                                file_handle,
-                                mmap_mode=None)
-
-    def _open_pickle(self, file_handle):
-        return BytesIO(read_zfile(file_handle))
-
-
 ###############################################################################
 # Utility functions
 
-def dump(value, filename, compress=0, cache_size=100, protocol=None):
-    """Fast persistence of an arbitrary Python object into one or multiple
-    files, with dedicated storage for numpy arrays.
+def dump(value, filename, compress=0, protocol=None, cache_size=0):
```

Comment:
Use cache_size=None by default. `cache_size is None` is a better way of making sure that the user has explicitly set it in its code.
Users could potentially have code like this
```python
dump(..., cache_size=0)
```

and they wouldn't get any DeprecationWarning.